lib/ansible/modules/windows/win_domain.ps1

```diff
@@ -51,31 +62,41 @@ Catch { }
 If(-not $forest) {
     $result.changed = $true
 
-    If(-not $check_mode) {
-        $sm_cred = ConvertTo-SecureString $safe_mode_admin_password -AsPlainText -Force
-
-        $install_forest_args = @{
-            DomainName=$dns_domain_name;
-            SafeModeAdministratorPassword=$sm_cred;
-            Confirm=$false;
-            SkipPreChecks=$true;
-            InstallDns=$true;
-            NoRebootOnCompletion=$true;
-        }
-        if ($database_path) {
-            $install_forest_args.DatabasePath = $database_path
-        }
-        if ($sysvol_path) {
-            $install_forest_args.SysvolPath = $sysvol_path
-        }
-        if ($domain_netbios_name) {
-            $install_forest_args.DomainNetBiosName = $domain_netbios_name
-        }
-        
-        $iaf = Install-ADDSForest @install_forest_args
+    $sm_cred = ConvertTo-SecureString $safe_mode_admin_password -AsPlainText -Force
+
+    $install_params = @{
+        DomainName=$dns_domain_name;
+        SafeModeAdministratorPassword=$sm_cred;
+        Confirm=$false;
+        SkipPreChecks=$true;
+        InstallDns=$true;
+        NoRebootOnCompletion=$true;
+        WhatIf=$check_mode;
+    }
+    if ($database_path) {
+        $install_params.DatabasePath = $database_path
+    }
+    if ($sysvol_path) {
+        $install_params.SysvolPath = $sysvol_path
+    }
+    if ($domain_netbios_name) {
+        $install_params.DomainNetBiosName = $domain_netbios_name
+    }
+    if ($create_dns_delegation -ne $null) {
+        $install_params.CreateDnsDelegation = $create_dns_delegation
```

Comment:
You've set `$create_dns_delegation` with a default value so it will never be null. Can just have this set in your initial hashtable unless you should change the arg definition instead.